mantidproject/mantid
# Muon results table logs display improvements

**Description of work.**

Improves the handling of logs in the results table of the new Muon GUI. An error is now shown if you try and choose a log that one of the results does not have. Also the value displayed is the time-averaged value only when the experiment is considered "running". For a simultaneous fit the time-average is computed for each input workspace and then the average of this value is taken.

The table also now has the column types set correctly: logs=X, parameters=Y.

It also fixes a bug with missing filtering single-period runs by status for all ISIS logs. The sample logs GUI had been doing this separately but the values from Python were the unfiltered ones. Note that there are still missing unit tests for `ISISRunLogs`. These will be added in a later issue as these are high-priority fixes.

**To test:**

Try out various fits and check different combinations of logs and results to check the column is displaying sensible results. Check the column types are as expected and they can be plotted.

Check out mixing results with different logs (from different instruments for example) and check you get an error. 

Fixes #25825 

*This does not require release notes* because **the new Muon work will be announced as one item.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
